src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressRing/ProgressRing.cs

```diff
@@ -244,8 +244,8 @@ private void SetAnimatedVisualPlayerSource(bool force = false)
 					}
 					else
 					{
-						_currentSourceUri = indeterminateSource == null ? FeatureConfiguration.ProgressRing.ProgressRingAsset : indeterminateSource.UriSource;
-						_player.Source.UriSource = _currentSourceUri;
+						_currentSourceUri = (indeterminateSource == null && indeterminateSource is not IAnimatedVisualSourceWithUri) ? FeatureConfiguration.ProgressRing.ProgressRingAsset : (indeterminateSource as IAnimatedVisualSourceWithUri)!.UriSource;
+						(_player.Source as IAnimatedVisualSourceWithUri)!.UriSource = _currentSourceUri;
```

Comment:
The null suppression here doesn't feel safe, and per the current logic, the `indeterminateSource is not IAnimatedVisualSourceWithUri` check is redundant. Because if it was null, then the second part of the `&&` will be always true.

Reply:
@Youssef1313 
good catch for the condition, should be || i will fix it now. I add the condition to make the null suppression safer,

Reply:
@Robert-Louis The `||` would be correct, but https://github.com/unoplatform/uno/pull/12136#discussion_r1185077745 is even simpler IMO